package NDaemonLog;


/*
  If change this file run antirobot/scripts/gen_desc/run.sh
*/


message TRecord {
    optional string verdict = 1;
    optional float matrixnet = 2;
    optional string req_type = 3;
    optional string balancer_ip = 4;
    optional string spravka_ip = 5;
    optional string ident_type = 6;
    optional string user_ip = 7;
    optional string event_unixtime = 8;
    optional uint32 web_req_count = 9;
    optional string req_id = 10;
    optional bool random_captcha = 11;
    optional string partner_ip = 12;
    optional bool wizard_error = 13;
    optional string yandexuid = 14;
    optional string cacher_host = 15;
    optional string processor_host = 16;
    optional uint32 missed = 17;
    optional string icookie = 18;
    optional string service = 19;
    optional bool may_ban = 20;
    optional bool can_show_captcha = 21;
    optional float threshold = 22;
    optional string req_url = 23;
    optional string ip_list = 24;
    optional string block_reason = 25;
    optional bool cacher_blocked = 26;
    optional string cacher_block_reason = 27;
    map<string, float> exp_formulas = 28;
    optional string ja3 = 29;
    optional float robotness = 30;
    optional string host = 31;
    optional float suspiciousness = 32;
    optional string user_agent = 33;
    optional string referer = 34;
    optional bool cbb_whitelist = 35;
    repeated string rules = 36;
    repeated string mark_rules = 37;
    optional string ja4 = 38;
    optional string req_group = 39;
    repeated string prev_rules = 40;
    repeated string cbb_ban_rules = 41;
    repeated string prev_cbb_ban_rules = 42;
    optional float spravka_ignored = 43;
    optional float force_ban = 44;
    repeated string cbb_mark_rules = 45;
    optional bool catboost_whitelist = 47;
    optional string unique_key = 48;
    repeated string cbb_checkbox_blacklist_rules = 49;
    repeated string cbb_can_show_captcha_rules = 50;
    optional bool degradation = 51;
    optional string p0f = 52;
    repeated uint32 experiments_test_id = 53;
    optional string hodor = 54;
    optional string hodor_hash = 55;
    optional uint64 lcookie = 56;
    optional string jws_state = 57;
    optional float cacher_formula_result = 58;
    optional bool ban_source_ip = 59;
    optional fixed64 mini_geobase_mask = 60;
    map<string, float> cacher_exp_formulas = 61;
    optional bool matrixnet_fallback = 62;
    optional string jws_hash = 63;
    optional string yandex_trust_state = 64;
    optional bool valid_spravka_hash = 65;
    optional bool valid_autoru_tamper = 66;
    optional bool ban_fw_source_ip = 67;
};

message TCacherRecord {
    optional string unique_key = 1;
    // optional string timestamp = 2;
    optional string verdict = 3;
    optional string req_type = 4;
    optional string balancer_ip = 5;
    optional string spravka_ip = 6; 
    optional string ident_type = 7;
    optional string user_ip = 8;
    optional string req_id = 9;
    optional bool random_captcha = 10;
    optional string partner_ip = 11;
    optional string yandexuid = 12; 
    optional string cacher_host = 13;
    optional string icookie = 14;
    optional string service = 15;
    optional bool may_ban = 16;
    optional bool can_show_captcha = 17;
    optional string req_url = 18;
    optional string ip_list = 19;
    optional string block_reason = 20;
    optional bool cacher_blocked = 21;
    optional string cacher_block_reason = 22;
    optional string ja3 = 23;
    optional float robotness = 24;
    optional string host = 25;
    optional float suspiciousness = 26;
    optional string user_agent = 27;
    optional string referer = 28;
    optional bool cbb_whitelist = 29;
    optional string ja4 = 30;
    optional string req_group = 31;
    repeated string cbb_mark_rules = 32;
    repeated string cbb_ban_rules = 33;
    optional float spravka_ignored = 34;
    optional bool catboost_whitelist = 35;
    repeated string cbb_checkbox_blacklist_rules = 36;
    repeated string cbb_can_show_captcha_rules = 37;
    optional bool degradation = 38;
    optional string p0f = 39;
    repeated uint32 experiments_test_id = 40;
    optional string hodor = 41;
    optional string hodor_hash = 42;
    optional uint64 lcookie = 43;
    optional string jws_state = 44;
    optional float cacher_formula_result = 45;
    optional bool ban_source_ip = 46;
    optional fixed64 mini_geobase_mask = 47;
    optional string jws_hash = 48;
    optional string yandex_trust_state = 49;
    optional bool valid_spravka_hash = 50;
    optional bool valid_autoru_tamper = 51;
    optional bool ban_fw_source_ip = 52;
    optional int64 timestamp = 53;
    optional string uuid = 54;
};

message TProcessorRecord {
    optional string unique_key = 1;
    // optional string timestamp = 2;
    optional float matrixnet = 3;
    optional uint32 web_req_count = 4;
    optional bool wizard_error = 5;
    optional string processor_host = 6;
    optional uint32 missed = 7;
    optional float threshold = 8;
    map<string, float> exp_formulas = 9;
    repeated string rules = 10;
    repeated string mark_rules = 11;
    repeated string prev_rules = 12;
    repeated string prev_cbb_ban_rules = 13;
    optional float force_ban = 14;
    map<string, float> cacher_exp_formulas = 15;
    optional bool matrixnet_fallback = 16;
    optional int64 timestamp = 17;
};
